lib/rentlinx/errors.rb

```diff
@@ -7,4 +7,37 @@ class UnexpectedAttributes < RentlinxError
 
   class InvalidTypeParam < RentlinxError
   end
+
+  class InvalidObject < RentlinxError
+    def initialize(type)
+      super("The given #{type} is missing required attributes and cannot be posted.")
+    end
+  end
+
+  class HTTPError < RentlinxError
+  end
+
+  class BadRequest < HTTPError
+    def initialize
+      super('The request sent to the server was invalid.')
+    end
+  end
+
+  class NotFound < HTTPError
+    def initialize
+      super('The item you requested could not be found on the remote server.')
```

Comment:
The response object is generally pretty helpful. Some request libraries will even include the requested parameters as part of the response object (I'm not sure if the one being used here does).